File: c/cz/mg/vulkan/PFN_vkCreateSemaphore.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCreateSemaphore_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pSemaphore, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateSemaphore f = (PFN_vkCreateSemaphore)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkSemaphoreCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkSemaphore*)jniLongToPointer(pSemaphore))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkCreateSemaphore.java
package cz.mg.vulkan;

public class PFNvkCreateSemaphore extends VkFunctionPointer {
    public PFNvkCreateSemaphore() {
    }

    public PFNvkCreateSemaphore(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateSemaphore(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCreateSemaphore(long value) {
        setValue(value);
    }

    public PFNvkCreateSemaphore(VkInstance instance) {
        super(instance, new VkString("vkCreateSemaphore"));
    }

    public void call(VkDevice device, VkSemaphoreCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkSemaphore pSemaphore, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pSemaphore != null ? pSemaphore.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    protected static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pSemaphore, long rval);
}
